astropy/modeling/tests/test_parameters.py

```diff
@@ -127,226 +125,255 @@ def test_parameter_operators():
     assert -par == -num
     assert abs(par) == abs(num)
 
-
-class TestParameters:
-
-    def setup_class(self):
-        """
-        Unit tests for parameters
-
-        Read an iraf database file created by onedspec.identify.  Use the
-        information to create a 1D Chebyshev model and perform the same fit.
-
-        Create also a gausian model.
-        """
-        test_file = get_pkg_data_filename('data/idcompspec.fits')
-        f = open(test_file)
-        lines = f.read()
-        reclist = lines.split("begin")
-        f.close()
-        record = irafutil.IdentifyRecord(reclist[1])
-        self.icoeff = record.coeff
-        order = int(record.fields['order'])
-        self.model = models.Chebyshev1D(order - 1)
-        self.gmodel = models.Gaussian1D(2, mean=3, stddev=4)
-        self.linear_fitter = fitting.LinearLSQFitter()
-        self.x = record.x
-        self.y = record.z
-        self.yy = np.array([record.z, record.z])
-
-    def test_set_slice(self):
-        """
-        Tests updating the parameters attribute with a slice.
-
-        This is what fitters internally do.
-        """
-
-        self.model.parameters[:] = np.array([3, 4, 5, 6, 7])
-        assert (self.model.parameters == [3., 4., 5., 6., 7.]).all()
```

Comment:
This does not work any more.

Reply:
It no longer is meant to work. Slicing assignment directly to the model.parameter attribute will not update the parameters without calling model._array_to_parameters().  The new api is to assign a new array to model.parameters which will do such an update. If slicing is required, it must be followed by the _array_to_parameters method call.

Reply:
should probably be mentioned in the change log